File: Algorithms/Sorting/python/bubbleSort.py


def sort(arr):
    n = len(arr)
    
    for i in range(n):
        swapped = False
        
        for j in range (0,n-i-1):
            
            if arr[j] > arr[j+1]:
                
                arr[j],arr[j+1] = arr[j+1], arr[j]
                swapped = True
                
        if swapped == False:
            break
        
    print(arr)

arr = [12,0x3,1990,28,0x4,2008]

sort(arr)

File: Algorithms/Sorting/cpp/bubbleSort.cpp
#include <iostream>
#include <vector>

void swap(int& a, int& b) {
    int temp = a;
    a = b;
    b = temp;
}

void bubbleSort(std::vector<int>& arr) {
    int n = arr.size();

    for (int i = 0; i < n - 1; ++i) {
        bool swapped = false;

        for (int j = 0; j < n - i - 1; ++j) {
            if (arr[j] > arr[j + 1]) {
                swap(arr[j], arr[j + 1]);
                swapped = true;
            }
        }

        if (swapped == false) {
            break;
        }
    }
}

int main() {
    std::vector<int> arr = {12, 0x3, 1990, 28, 0x4, 2008};

    bubbleSort(arr);

    std::cout << "Sorted array: ";
    for (int num : arr) {
        std::cout << num << " ";
    }
    std::cout << std::endl;

    return 0;
}
